### ISSUE TITLE
Apps update with cli resulting in an error

### ISSUE THREAD

**Issue by mholasek** [rancher/rancher#21253]
Update an installed app via rancher cli using a project catalog resulting in an error.
Used command:
```bash
rancher app upgrade --set image.tag=v0.1.15 appName 0.1.2
```

Error:
```
FATA[0001] version 0.1.2 is not valid
```
The helm chart version 0.1.2 exists.

Installation via cli was possible:
```bash
rancher app install --set image.tag=0.1.15 chartName appName
```
---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.2.4` UI: `v2.2.70` |
|Route|`containers.index`|

**Comment by loganhz** [comment]
https://github.com/rancher/rancher/issues/21282 is blocked by this issue

**Comment by deniseschannon** [comment]
@loganhz please see if someone on your team can fix this.

**Comment by dramich** [comment]
available in v2.3.0-rc3

**Comment by sowmyav27** [comment]
Verified on `v2.2.7`'s rancher CLI.
Steps:

- Deploy an rke cluster
- Add a project level catalog
- Through the rancher CLI, install mysql - version - 0.3.7 (from the added catalog)
- Upgrade the app to version - 0.3.8, an error will be seen.

```
./rancher app upgrade a-pdps4 0.3.8
FATA[0001] version 0.3.8 is not valid   
```

Verified in `v2.3.0-rc3` and `master-head` (latest)
- Deploy an rke cluster
- Add a project level catalog
- Through the rancher CLI, install mysql - version - 0.3.7 (from the added catalog)
- Upgrade the app to version - 0.3.8
- No error is seen on Rancher CLI. On the Rancher UI, app is upgraded to version 0.3.8.
